std/array.d

```diff
@@ -126,12 +126,33 @@ if (isIterable!Range && !isAutodecodableString!Range && !isInfinite!Range)
         auto result = (() @trusted => uninitializedArray!(Unqual!E[])(length))();
 
         // Every element of the uninitialized array must be initialized
-        size_t i;
-        foreach (e; r)
+        size_t cnt; //Number of elements that have been initialized
+        try
         {
-            emplaceRef!E(result[i], e);
-            ++i;
+            foreach (e; r)
+            {
+                emplaceRef!E(result[cnt], e);
+                ++cnt;
+            }
+        } catch (Exception e)
+        {
+            //https://issues.dlang.org/show_bug.cgi?id=22185
+            //Make any uninitialized elements safely destructible.
+            foreach(ref elem; result[cnt..$])
+            {
+                emplaceRef!E(elem, E.init);
```

Comment:
`emplaceRef!E(elem);` uses cheap `emplaceInitializer`: https://github.com/dlang/druntime/blob/9d99b1ccd90d1e66c31e27cc71b8a266612aa14a/src/core/internal/lifetime.d#L23

Reply:
Ok lets use that instead